### ISSUE TITLE
Clarification: What does Reputation Represent, Anyway

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#5110]
**Is your feature request related to a problem? Please describe.**
Right now there is some degree of confusion as to what Reputation represents, or what it is supposed to represent. Is it simply a legal standing with the government? Does it represent how well liked the player is? Local popularity? faction-wide popularity? A simple metric for hostile vs friendly?

The intent of this issue is to clarify exactly what Reputation represents so that players and creators alike and use it in a manner that makes sense. 

**Related Issue Links**
https://github.com/endless-sky/endless-sky/pull/5106

**Describe the solution you'd like**
A clear and precise description of what reputation is supposed to represent.
If appropriate, a specification for how the reputation system could be adjusted to better reflect what we are trying to do.

**Thoughts:**
_1. Reputation as legal standing:_
The idea that reputation represents the player's legal standing is somewhat persuasive, as many of the big changes that happen to it are in fixed "reset to X" values that happen to indicate cease fires, ends of cease fire, etc. These then basically force the government into and out of hostility with the player.

On the other hand, most legal systems today consider murder to be a "arrest immediately" sort of offense, not just a "people like you a bit less" sort of thing. Likewise, most governments do not consider wantonly killing people in jurisdictions they disagree with to be a way to get back on the right side of the law. For that matter, doing lots of work for them is generally not going to earn anyone a pass on murder or grand theft.

_2. Reputation as Popularity:_
This idea is what the term implies, to me. This is reinforced by the fact that destroying someone's ship just makes the local people hostile, but doesn't necessarily make the whole government hostile. So if someone is particularly well liked, (as in, they are very useful) they can get away with pillaging the occasional ship without too much problem.

On the other hand, sheer popularity shouldn't be significantly affected by cease fires and the like. It is perfectly possible to be well liked while still being an enemy combatant.

**Comment by Naminson** [comment]
I think it depends on government. Obviously, if you do something bad to a government, they will become hostile. But the exact motives of someone who decides to attack is going to vary. I don't think that the Navy retaliates based on their opinion of you, since that's usually out of character for them. They leave that up to parliament, for the most part. However, the merchants would try and disable you based on their opinion because they are less of a faction and more of a general group of people. I think that the example for any definition of reputation is a faction that is forced to attack you, despite liking you, or the other way around.

That being said, I think the best answer I can think of is to say that your reputation is that if you take every factor the average person knows about, then your reputation is whatever conclusion the average person will come to. The most common case being direct aggression vs. friendly chatter, but there could be other examples in the future, even without engine changes. That definition isn't perfect, but it's the most specific I can think of before you prevent something potentially good from existing.

Of course, it also doesn't mean we shouldn't have more specific definitions of what those factors are. However, that's something that would merit it's own discussion for every faction.

**Comment by meiwud334** [comment]
You are over thinking this. reputation is a game mechanic that keeps track of the players generall behaviour. It allows for some crimes because this is a game but continues misbehaviour leads to hostility. The game should give playees options to qctively influence in small ways theie reputation to allow more sandbox freedom.

**Comment by Zitchas** [comment]
It occurs to me that it might be useful / interesting to split reputation into two attributes:

Hostility & Reputation.

**Hostility**
This would be a set value with 4 options: -1 is variable (aka use reputation), 0 is always hostile, 1 is always neutral, and 3 is always friendly. In this case "always" means exactly that. Anything short of directly acting on ships of that specific government will not affect its status. The -1 option is included for factions that don't have overarching governments or command structures and thus use the reputation value instead.
The main use of the hostility attribute would be for things like controlling allegiances during war, cease-fires, etc. For example, once you are flying Free World colors, the Republic is set to `hostility 0` and the Free World is set to `hostility 3`, which means that it will always be hostile to you. During cease-fires it could be set to `hostility 1`.
The secondary use for this would be as a secondary control on access to planets. Planets of a faction that are locked to "hostile" don't let you land, friendly ones always let you land, etc.

**Reputation**
This value represents how well liked you are by the faction. Doing jobs of all (positive to them) kinds, killing their enemies, etc would all positively affect reputation to varying degrees. For factions that do not have some overriding war or other reason locking them into a specific hostility or lack thereof, reputation is what dictates if the government attacks you or not.

The more interesting part is that reputation would affect priorities within combat as well. Say you and a bunch of Free World NPCs are attacking a Republic fleet. Due to the war, the republic fleet is locked in "hostile". But you have spent a lot of time building up a high positive reputation with the republic prior to the war. As a result, the combat AI starts to prioritize other ships over you when picking targets. This isn't a blanket immunity, rather it is a form of weighting. If the Republic cruiser has two Free World ships that are about equally distant from it, and one is a known upstanding citizen who has done lots of jobs and stuff; whereas the other Free World ship had been known to pick off the occasional merchant ship in Republic space (but did enough regular jobs to stay positive in reputation)... Who do you think the Republic ship should prioritize? The almost criminal one, of course.

Perhaps higher degrees of reputation might affect the likelyhood of ships taking opportunistic shots against you when they are focused on other targets.

Likewise, in combat against factions that accept bribes, one's reputation could be used as a variable for determining how much they charge in bribes to leave you alone, or let you land. In some case, perhaps hostile worlds that normally do not accept bribes might be willing to accept them from players with sufficiently high reputation.

This separates out "Forced Hostility" from the player's actual "Reputation" with a particular group, which means that reputation doesn't need to be forcefully reset so often, and thus actually does start to mean their reputation and not just be a variable for forcing/removing hostility.

**Comment by Amazinite** [comment]
What can we not already do by manipulating the current government attitudes that we could do with this split reputation idea?
Want them to always be hostile? Don't allow you to increase your reputation with them by killing their enemies. 
Want them to always be friendly? Drop their drop their "penalty for"s to 0.
It sounds to me like what's really being asked is for reputation to have more effects than just on hostility (e.g. affect bribes, attack priority).

**Comment by Zitchas** [comment]
Yeah, that is definitely part of it. But part of it is that some people seem to like having reputation actually reflect on they are perceived by that government; not just have it be an arbitrary value that gets flipped back and forth. Right now, reputation doesn't do much. About its only real value is being something that can be farmed to allow the capture of some otherwise impossible to get ships. 

Right now, it doesn't really matter if someone gets their reputation to any particular amount, because it is meaningless and will in all likelyhood get reset by some random mission. And the mission resets usually either are hard resets to specific values, or arbitrary "starts at -1000 so you have no hope of ever recovering it". For things like that, why not leave reputation out of it and just use a simple attribute to say whether or not they are hostile, and allow reputation to do something different that suits its variable nature?

**Comment by Amazinite** [comment]
Alternatively, with #3128, we can "save" your reputation with a government by doing something like `"old pirate reputation" = "reputation: Pirate"` then later down the road do the reverse `"reputation: Pirate" = "old pirate reputation"` to reset it.
